spec/support_specs/logstash_logging_spec.rb

```diff
@@ -0,0 +1,95 @@
+require 'rails_helper'
+
+RSpec.describe LogstashLogging do
+  include TestHelpers::CaptureStdoutHelper
+
+  before {
+    @lograge = Struct.new(:enabled, :base_controller_class, :custom_options, :formatter).new
```

Comment:
Well, since you're asking! I do have many opinions on this.

I prefer to avoid let/ivar/before altoghether, in the case of this test I'd do:

```rb
describe 'ClockworkCheck' do
  it 'logs it has run to the default Rails logger' do
    logger = stub_rails_logger

    ClockworkCheck.new.perform
    
    expect(logger.string).not_to be_blank
  end

  it 'says "clockwork is running..."' do
    logger = stub_rails_logger

    ClockworkCheck.new.perform

    expect(logger.string).to match(/clockwork is running\.\.\./)
  end
end

def stub_rails_logger
  stringio_output = StringIO.new
  stringio_logger = Logger.new(stringio_output)
  allow(Rails).to receive(:logger).and_return(stringio_logger)
  stringio_output
end
```


https://thoughtbot.com/blog/four-phase-test
https://thoughtbot.com/blog/lets-not

Reply:
Well, since you're asking! I do have many opinions on this.

I prefer to avoid let/ivar/before altoghether, in the case of this test I'd do:

```rb
describe 'ClockworkCheck' do
  it 'logs it has run to the default Rails logger' do
    logger = stub_rails_logger

    ClockworkCheck.new.perform
    
    expect(logger.string).not_to be_blank
  end

  it 'says "clockwork is running..."' do
    logger = stub_rails_logger

    ClockworkCheck.new.perform

    expect(logger.string).to match(/clockwork is running\.\.\./)
  end
end

def stub_rails_logger
  stringio_output = StringIO.new
  stringio_logger = Logger.new(stringio_output)
  allow(Rails).to receive(:logger).and_return(stringio_logger)
  stringio_output
end
```


https://thoughtbot.com/blog/four-phase-test
https://thoughtbot.com/blog/lets-not

Reply:
I'd prefer to follow @tijmenb's suggestion here unless you have objections?